### ISSUE TITLE
Type hint for `configure_scope` suggests it can return `None` but it should not

### ISSUE THREAD

**Issue by rafrafek** [getsentry/sentry-python#1799]
### How do you use Sentry?

Sentry Saas (sentry.io)

### Version

1.12.1

### Steps to Reproduce

```python3
import sentry_sdk
with sentry_sdk.configure_scope() as scope:
    pass
```

### Expected Result

`configure_scope` should always return `ContextManager` and should never return `None`.

Instead of:
```python3
-> (ContextManager | None)
```
should always return:
```python3
-> (ContextManager)
```

### Actual Result

Pylance complains:

```python3
(function) configure_scope(callback: ((Scope) -> None) | None = None) -> (ContextManager | None)
```
```error
Object of type "None" cannot be used with "with"
```
_Pylance [reportOptionalContextManager](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportOptionalContextManager)_

<img width="895" alt="image" src="https://user-images.githubusercontent.com/23004737/208939543-11d3722c-37f1-4455-bf7b-63456e770804.png">

### Workaround

```python3
import sentry_sdk
scope_ctx = sentry_sdk.configure_scope()
if scope_ctx is not None:
    with scope_ctx as scope:
        pass
```

or using `:=`:

```python3
import sentry_sdk
if (scope_ctx := sentry_sdk.configure_scope()) is not None:
    with scope_ctx as scope:
        pass
```

**Comment by szokeasaurusrex** [comment]
Hello @rafrafek, I tried to reproduce your error, but with the code you provided, my type hint shows up correctly, as you can see in the screenshot below:

![Image](https://github.com/getsentry/sentry-python/assets/7881302/5021bed7-12b2-46a3-9333-9b35b4127e05)


Since everything appears to be working correctly on my end, I will close the issue for now.

However, if you are still experiencing this problem, please feel free to reopen the issue. And, if you reopen the issue, please also provide us with more details regarding your editor setup, so that we can reproduce the problem you found.

**Comment by rafrafek** [comment]
Thanks for looking into it! 

I think it was fixed in [1.17.0](https://github.com/getsentry/sentry-python/releases/tag/1.17.0) thanks to `Fix: Rename MYPY to TYPE_CHECKING` #1934 so now Pylance/Pyright can detect overloads properly.

**Comment by szokeasaurusrex** [comment]
Great! Thank you for reporting the issue, and glad to hear it is working now
